src/runtime/GlobalObjectBuiltinBigInt.cpp

```diff
@@ -136,12 +140,38 @@ Value builtinBigIntToString(ExecutionState& state, Value thisValue, size_t argc,
     return S->toString(radix);
 }
 
-Value builtinBigIntValueOf(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+static Value builtinBigIntValueOf(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
 {
     RESOLVE_THIS_BINDING_TO_BIGINT(S, BigInt, valueOf);
     return Value(S);
 }
 
+static Value builtinBigIntToLocaleString(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    RESOLVE_THIS_BINDING_TO_BIGINT(thisObject, BigInt, toLocaleString);
+
+#if defined(ENABLE_ICU) && defined(ENABLE_INTL)
+    Value locales = argc > 0 ? argv[0] : Value();
+    Value options = argc > 1 ? argv[1] : Value();
+    Object* numberFormat = IntlNumberFormat::create(state, state.context(), locales, options);
+    auto result = IntlNumberFormat::format(state, numberFormat, thisObject->toString());
+
+    return new UTF16String(result.data(), result.length());
+#else
+
+    ObjectGetResult toStrFuncGetResult = state.context()->globalObject()->bigIntProxyObject()->get(state, ObjectPropertyName(state.context()->staticStrings().toString));
+    if (toStrFuncGetResult.hasValue()) {
+        Value toStrFunc = toStrFuncGetResult.value(state, thisObject);
+        if (toStrFunc.isCallable()) {
+            // toLocaleString() ignores the first argument, unlike toString()
+            return Object::call(state, toStrFunc, thisObject, 0, argv);
+        }
+    }
+    ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().BigInt.string(), true, state.context()->staticStrings().toLocaleString.string(), ErrorObject::Messages::GlobalObject_ToLocaleStringNotCallable);
+    RELEASE_ASSERT_NOT_REACHED();
+#endif
```

Comment:
Dummy return statement is necessary at this point because `builtinBigIntToLocaleString` function should return `Value` type. Otherwise a compile error would occur for non-icu build environment.

Reply:
you are right!